packages/client/src/utils/buttonActions.js

```diff
@@ -429,12 +440,14 @@ export const enrichButtonActions = (actions, context) => {
                   buttonContext.push(result)
                   const newContext = { ...context, actions: buttonContext }
 
-                  // Enrich and call the next button action
-                  const next = enrichButtonActions(
-                    actions.slice(i + 1),
-                    newContext
-                  )
-                  resolve(await next())
+                  // Enrich and call the next button action if there is more than one action remaining
+                  if (actions.length > 1) {
```

Comment:
We don't actually need to add this check - `enrichButtonActions` won't do anyting when an empty array is passed in.